alpha/apps/kaltura/lib/KSecureEntryHelper.class.php

```diff
@@ -252,13 +252,38 @@ protected function isFlavorParamsAllowed($flavorParamsId)
 
 	public function updateDeliveryAttributes(DeliveryProfileDynamicAttributes $deliveryAttributes)
 	{
-		$actionList = $this->getActionList(RuleActionType::LIMIT_DELIVERY_PROFILES);
-		if ($actionList)
+		$limitDeliveryProfilesActionList = $this->getActionList(RuleActionType::LIMIT_DELIVERY_PROFILES);
```

Comment:
Please move the implementation into the action and run on all actions and call this new implementing method, for example applyDeliveryProfileDynamicAttributes().